synthesizer/event.cpp

```diff
@@ -370,5 +370,43 @@ void EventList::insert(const Event& e)
             }
       append(e);
       }
-}
 
+//---------------------------------------------------------
+//   class EventMap::fixupMIDI
+//---------------------------------------------------------
+
+void EventMap::fixupMIDI()
+      {
+      unsigned short nowPlaying[_highestChannel + 1][128 /* notes */] = { { 0 } };
+      auto it = begin();
+
+      while (it != end()) {
+            bool discard = false;
+
+            /* ME_NOTEOFF is never emitted, no need to check for it */
+            if (it->second.type() == ME_NOTEON) {
+                  unsigned short np = nowPlaying[it->second.channel()][it->second.pitch()];
+                  if (it->second.velo() == 0) {
+                        if (np == 0) {
```

Comment:
Too many unneeded braces for my taste

Reply:
OK. Some people at work insist on them, but I agree and will remove them (and see after the tests).

Reply:
And maybe using `if ((np == 0) || (--np > 0))` here? Not sure though whether this might undefined behavoir

Reply:
as per https://en.wikipedia.org/wiki/Sequence_point this should be clean, the `||` is a sequence point.Leave a comment

Reply:
Hmm, okay. I had a few debugging `fprintf(stderr, …);` in between, which is why I had the separate logic.